pandas/core/dtypes/cast.py

```diff
@@ -1474,7 +1474,12 @@ def find_common_type(types):
         return first
 
     if any(isinstance(t, ExtensionDtype) for t in types):
-        return np.object
+        for t in types:
+            if isinstance(t, ExtensionDtype):
+                res = t._get_common_type(types)
+                if res is not None:
+                    return res
+        return np.dtype("object")
```

Comment:
i like having this logic live here.

@TomAugspurger the "negotiation protocol" would affect what order we iterate over types?

Reply:
Dunno, but I doubt it. At the end of the day we have established behavior that order matters (I think).